internal/client.go

```diff
@@ -551,6 +552,12 @@ func NewClient(service workflowserviceclient.Interface, domain string, options *
 	if options != nil && options.Tracer != nil {
 		tracer = options.Tracer
 		contextPropagators = append(contextPropagators, NewTracingContextPropagator(zap.NewNop(), tracer))
```

Comment:
```suggestion
```
same as done below, otherwise this seems like using duplicate propagators.

Reply:
yep, definitely. that should not been here.